src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -226,6 +226,58 @@ namespace
         }
     };
 
+    struct DiffuseClosure
+    {
+        typedef OrenNayarBRDFInputValues InputValues;
+
+        struct Params
+        {
+            OSL::Vec3   N;
+        };
+
+        static const char* name()
+        {
+            return "diffuse";
+        }
+
+        static ClosureID id()
+        {
+            return DiffuseID;
+        }
+
+        static void register_closure(OSL::ShadingSystem& shading_system)
+        {
+            const OSL::ClosureParam params[] =
+            {
+                CLOSURE_VECTOR_PARAM(Params, N),
+                CLOSURE_FINISH_PARAM(Params)
+            };
+
+            shading_system.register_closure(name(), id(), params, 0, 0);
+
+            g_closure_convert_funs[id()] = &convert_closure;
+        }
+
+        static void convert_closure(
+            CompositeSurfaceClosure&    composite_closure,
+            const Basis3f&              shading_basis,
+            const void*                 osl_params,
+            const Color3f&              weight)
+        {
+            const Params* p = reinterpret_cast<const Params*>(osl_params);
+            InputValues* values =
+                composite_closure.add_closure<InputValues>(
+                    OrenNayarID,
+                    shading_basis,
+                    weight,
+                    p->N);
+
+            values->m_reflectance.set(1.0f);
+            values->m_reflectance_multiplier = 1.0;
```

Comment:
What's the type of `m_reflectance_multiplier`? If it's float (it should be I think) then replace `1.0` by `1.0f`.